### ISSUE TITLE
side colors don't show unless zoomed

### ISSUE THREAD

**Issue by jbholm** [talgalili/heatmaply#145]
Hi, 

When I am producing an .html file for my heatmap, the column side colors do not show when the heatmap is at default zoom setting (zoomed all the way out). The colors then show only after I manually zoom in on a portion of the column color bar. 

In the same way, if I produce a static .png file, the column side colors are not produced. 

Any help is appreciated! Details below:

Package: heatmaply
Type: Package
Title: Interactive Cluster Heat Maps Using 'plotly'
Version: 0.14.0
Date: 2018-01-20

heatmaply(t(svm_relabund_sort[,5:30]), Rowv = FALSE,  Colv=FALSE, col_side_colors = svm_relabund_sort$CST, file="heatmaply_cst.html", labCol = svm_relabund_sort$sampleID, cexRow = 0.6, cexCol = 0.5, col=rainbow(50,start=1/6,end=0), key=FALSE, titleX = FALSE, titleY = FALSE, column_text_angle = 90, hide_colorbar=TRUE, colorbar_len=0, anim_duration=0)

**Comment by Alanocallaghan** [comment]
I can't reproduce the error, could you post a screenshot?

**Comment by jbholm** [comment]
![zoomed_area](https://user-images.githubusercontent.com/17168205/35241175-a0e7f588-ff83-11e7-9752-195e9aacaea7.jpeg)
![chosen_area_to_zoom](https://user-images.githubusercontent.com/17168205/35241181-a454acac-ff83-11e7-8dd6-21a21500a972.jpeg)
![zoomed_out](https://user-images.githubusercontent.com/17168205/35241185-a6591754-ff83-11e7-82e4-29b1f8b0592a.jpeg)

**Comment by Alanocallaghan** [comment]
What is the dimension of your matrix, ie `dim(svm_relabund_sort)`?

**Comment by jbholm** [comment]
dim(svm_relabund_sort)
[1] 5369 1913

But I plot only columns 5:30 (note that I had to transpose the matrix for it to plot how I like it. So there are 30 rows and 5369 columns in the figures.

**Comment by Alanocallaghan** [comment]
#53

**Comment by Alanocallaghan** [comment]
The side colours are shown, but the gap between them prevents them being visible, since they are plotted so close. This is clear since hovertext still shows for some cells.

This is a known issue with ggplot2 and plotly. You can get around this by setting `plot_method = "plotly"` in the call to `heatmaply()`. This also greatly improves the speed of producing the plot and shouldn't have other major issues.

Try this:

```r
heatmaply(
    t(svm_relabund_sort[,5:30]), 
    Rowv = FALSE, 
    Colv=FALSE, 
    col_side_colors = svm_relabund_sort$CST, 
    file="heatmaply_cst.html", 
    labCol = svm_relabund_sort$sampleID, 
    cexRow = 0.6, 
    cexCol = 0.5, 
    col=rainbow(50,start=1/6,end=0), 
    key=FALSE, 
    titleX = FALSE, 
    titleY = FALSE, 
    column_text_angle = 90, 
    hide_colorbar=TRUE, 
    colorbar_len=0, 
    anim_duration=0,
    plot_method = "plotly")
```

**Comment by Alanocallaghan** [comment]
If this solves your issue I will close this as it is essentially a duplicate of #53. Cheers

**Comment by jbholm** [comment]
It worked!! Go ahead and close it down. and THANK YOU!

**Comment by Alanocallaghan** [comment]
Cheers, glad to help
